app/scripts/controllers/preferences-controller.ts

```diff
@@ -149,125 +159,289 @@ export type PreferencesControllerState = {
+    anonymous: true,
+  },
+  knownMethodData: {
+    persist: true,
+    anonymous: false,
+  },
+  currentLocale: {
+    persist: true,
+    anonymous: true,
+  },
+  identities: {
+    persist: true,
+    anonymous: false,
+  },
+  lostIdentities: {
+    persist: true,
+    anonymous: false,
+  },
+  forgottenPassword: {
+    persist: true,
+    anonymous: true,
+  },
+  preferences: {
+    persist: true,
+    anonymous: true,
+  },
+  ipfsGateway: {
+    persist: true,
+    anonymous: false,
+  },
+  isIpfsGatewayEnabled: {
+    persist: true,
+    anonymous: false,
+  },
+  useAddressBarEnsResolution: {
+    persist: true,
+    anonymous: true,
+  },
+  ledgerTransportType: {
+    persist: true,
+    anonymous: true,
+  },
+  snapRegistryList: {
+    persist: true,
+    anonymous: false,
+  },
+  theme: {
+    persist: true,
+    anonymous: true,
+  },
+  snapsAddSnapAccountModalDismissed: {
+    persist: true,
+    anonymous: false,
+  },
+  useExternalNameSources: {
+    persist: true,
+    anonymous: false,
+  },
+  useTransactionSimulations: {
+    persist: true,
+    anonymous: true,
+  },
+  enableMV3TimestampSave: {
+    persist: true,
+    anonymous: true,
+  },
+  useExternalServices: {
+    persist: true,
+    anonymous: false,
+  },
+  textDirection: {
+    persist: true,
+    anonymous: false,
+  },
+};
 
-  /**
-   *
-   * @param opts - Overrides the defaults for the initial state of this.store
-   * @property messenger - The controller messenger
-   * @property initState The stored object containing a users preferences, stored in local storage
-   * @property initState.useBlockie The users preference for blockie identicons within the UI
-   * @property initState.useNonceField The users preference for nonce field within the UI
-   * @property initState.featureFlags A key-boolean map, where keys refer to features and booleans to whether the
-   * user wishes to see that feature.
-   *
-   * Feature flags can be set by the global function `setPreference(feature, enabled)`, and so should not expose any sensitive behavior.
-   * @property initState.knownMethodData Contains all data methods known by the user
-   * @property initState.currentLocale The preferred language locale key
-   * @property initState.selectedAddress A hex string that matches the currently selected address in the app
-   */
-  constructor(opts: PreferencesControllerOptions) {
+export default class PreferencesController extends BaseController<
+  typeof controllerName,
+  PreferencesControllerState,
+  PreferencesControllerMessenger
+> {
+  /**
+   * Constructs a Preferences controller.
+   *
+   * @param options - the controller options
+   * @param options.networkConfigurationsByChainId - The network configurations
+   * @param options.initLangCode - The language code
+   * @param options.messenger - The controller messenger
+   * @param options.state - The initial controller state
+   */
+  constructor({
+    networkConfigurationsByChainId,
```

Comment:
Do you think we can get this with a messenger call, so we can remove this option?